src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,330 @@
+  [FiatAccountSchema.AccountNumber]: [
+    {
+      name: 'accountName',
+      label: i18n.t('fiatAccountSchema.accountName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
+    },
+    {
+      name: 'institutionName',
+      label: i18n.t('fiatAccountSchema.institutionName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^[0-9]{10}$/,
+      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
+    },
+  ],
+}
+
+// We need to compare the body collected from form to the Interface of the fiat account schema
+// This is a helper function that returns a dummy object of FiatAccountSchema to iterate over
+const getSchemaObjectByType = (fiatAccountSchema: FiatAccountSchema) => {
+  let newSchema: AccountNumber | undefined
+  switch (fiatAccountSchema) {
+    case FiatAccountSchema.AccountNumber:
+      newSchema = {
+        accountName: '',
+        institutionName: '',
+        accountNumber: '',
+        country: '',
+        fiatAccountType: FiatAccountType.BankAccount,
+      }
+      break
+    default:
+      newSchema = undefined
+  }
+
+  return newSchema
+}
+
+const FiatDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema, cicoQuote } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set<string>())
+  const inputRefs = useRef<string[]>([textValue])
+  const userCountry = useSelector(userLocationDataSelector)
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('fiatDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: Record<string, string> = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      const validatedBody = validateAndCompleteSchema(
+        body,
+        getSchemaObjectByType(fiatAccountSchema)
+      )
```

Comment:
This seems overly complicated for just adding two new values to the body, and it runs on all schemas when the additions are specific to AccountNumber.

You could just have a function like this
```js
const implicitFields = (fiatAccountSchema) => {
switch (fiatAccountSchema) {
    case FiatAccountSchema.AccountNumber:
      return { country: userCountry.countryCodeAlpha2, fiatAccountType: FiatAccountType.BankAccount }
      break
    default:
      return {}
  }
}
```

and then do this here

```js
const completeBody = {
...body,
...implicitFields(fiatAccountSchema)
}
```

Reply:
I'd actually prefer it if you added fields like `country` and `fiatAccountType` to SCHEMA_TO_FIELD_METADATA_MAP and have the metadata for those fields include functions to get the correct values and let the rest of the UI know not to display them. That way all of the schema is in one place and much easier to read and verify.